k8s/predelete.go

```diff
@@ -0,0 +1,86 @@
+package k8s
+
+import (
+	"context"
+	"tyk/tyk/bootstrap/pkg/constants"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ExecutePreDeleteOperations executes operations needed in pre-delete chart hook one by one.
+func (c *Client) ExecutePreDeleteOperations() error {
+	c.l.Info("Running pre delete hook")
+
+	if err := c.deleteSecret(c.appArgs.OperatorKubernetesSecretName, true); err != nil {
```

Comment:
Need to add a check if `c.appArgs.OperatorKubernetesSecretName` is not empty, then only delete secret. Otherwise it fails with error `resource name may not be empty`

Reply:
updated in e2281579a39339e6f0f93213519e24e904da832a